executor/common_linux.h

```diff
@@ -1734,6 +1732,141 @@ static int do_sandbox_namespace(void)
 }
 #endif
 
+#if SYZ_EXECUTOR || SYZ_SANDBOX_ANDROID_UNTRUSTED_APP
+#include <fcntl.h> // open(2)
+#include <grp.h> // setgroups
+#include <sys/xattr.h> // setxattr, getxattr
+
+#define AID_NET_BT_ADMIN 3001
+#define AID_NET_BT 3002
+#define AID_INET 3003
+#define AID_EVERYBODY 9997
+#define AID_APP 10000
+
+#define UNTRUSTED_APP_UID AID_APP + 999
+#define UNTRUSTED_APP_GID AID_APP + 999
+
+const char* SELINUX_CONTEXT_UNTRUSTED_APP = "u:r:untrusted_app:s0:c512,c768";
+const char* SELINUX_LABEL_APP_DATA_FILE = "u:object_r:app_data_file:s0:c512,c768";
+const char* SELINUX_CONTEXT_FILE = "/proc/thread-self/attr/current";
+const char* SELINUX_XATTR_NAME = "security.selinux";
+
+gid_t UNTRUSTED_APP_GROUPS[] = {UNTRUSTED_APP_GID, AID_NET_BT_ADMIN, AID_NET_BT, AID_INET, AID_EVERYBODY};
+size_t UNTRUSTED_APP_NUM_GROUPS = sizeof(UNTRUSTED_APP_GROUPS) / sizeof(UNTRUSTED_APP_GROUPS[0]);
+
+// Similar to libselinux getcon(3), but withouthe library dependency
+// and without any dynamic memory allocation.
+static int getcon(char* context, size_t context_size)
+{
+	int fd = open(SELINUX_CONTEXT_FILE, O_RDONLY);
+
+	if (fd < 0)
+		fail("getcon: Couldn't open %s", SELINUX_CONTEXT_FILE);
+
+	ssize_t nread = read(fd, context, context_size);
+
+	close(fd);
+
+	if (nread < 0)
+		fail("getcon: Failed to read from %s", SELINUX_CONTEXT_FILE);
+
+	// The contents of the context file MAY end with a newline
+	// and MAY not have a null terminator.  Handle this here.
+	if (context[nread - 1] == '\n')
+		context[nread - 1] = '\0';
+
+	return 0;
+}
+
+// see: man 3 setcon
+static int setcon(const char* context)
+{
+	char new_context[512];
+
+	// Attempt to write the new context
+	int fd = open(SELINUX_CONTEXT_FILE, O_WRONLY);
+
+	if (fd < 0)
+		fail("setcon: Could not open %s", SELINUX_CONTEXT_FILE);
+
+	ssize_t bytes_written = write(fd, context, strlen(context));
+	close(fd);
+
+	if (bytes_written != (ssize_t)strlen(context))
+		fail("setcon: Could not write entire context.  Wrote %zi, expected %zu", bytes_written, strlen(context));
+
+	// Validate the transition by checking the context
+	if (getcon(new_context, sizeof(new_context) != 0))
+		fail("setcon: Could not read context");
+
+	if (strcmp(context, new_context))
+		fail("setcon: Failed to change to %s, context is %s", context, new_context);
+
+	return 0;
+}
+
+// Similar to libselinux getfilecon(3), but without the library dependency
+// and without any dynamic memory allocation.
+static int getfilecon(const char* path, char* context, size_t context_size)
```

Comment:
Also `void`. And at that point, I suggest we just inline it in caller. It calls another function, so additional abstraction layer does not give us much.

Reply:
I'll leave this to the compiler to perform inlining.

This is NOT a performance critical path.

Logical separation into functions is useful, especially since I am (well, was) following the standard API interface to selinux.